private-templates-service/client/src/components/AllCards/TagList/TagList.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { Component } from "react";
+import Tags from "../../Common/Tags";
+import { Scroller, ScrollDirection } from "../../../utils/AllCardsUtil";
+import { TagsContainer } from "./styled";
+
+interface TagListProps {
+  tags: string[];
+  allowEdit: boolean;
+}
+class TagList extends Component<TagListProps> {
+  scroller: Scroller;
+  ref: any;
```

Comment:
type this please

Reply:
typing this introduces lots of problems: 
1) EventListeners for react components and for simple html elements (like divs) accept different callbacks (in terms of return type and arguments). So, in the places where we use ref to do the scrolling (to prevent the outer page to scroll with it) we need to use addEventListener while for other react components (like Version table in Preview modal) we simply use events (like **onWheel={..}**).